core/iwasm/compilation/aot_emit_aot_file.c

```diff
@@ -375,34 +378,64 @@ get_table_info_size(AOTCompContext *comp_ctx, AOTCompData *comp_data)
                                            comp_data->table_init_data_list,
                                            comp_data->table_init_data_count);
 }
-
+#if WASM_ENABLE_GC != 0
 static uint32
-get_func_type_size(AOTFuncType *func_type)
+get_func_type_size(AOTCompContext *comp_ctx, AOTFuncType *func_type)
+{
+    /* type flag + is_sub_final + parent_type_idx + param count + result count
+     * + types + ref_type_map_count */
+    if (comp_ctx->enable_gc) {
+        return sizeof(func_type->base_type.type_flag) + sizeof(uint16)
+               + sizeof(func_type->base_type.parent_type_idx)
+               + sizeof(func_type->param_count)
+               + sizeof(func_type->result_count) + func_type->param_count
+               + func_type->result_count
+               + sizeof(func_type->ref_type_map_count);
+    }
+    else {
+        /* type flag + is_sub_final + parent_type_idx + param count + result
+         * count + types */
+        return (uint32)sizeof(uint16) * 6 + func_type->param_count
+               + func_type->result_count;
+    }
+}
+#else
+static uint32
+get_func_type_size(AOTCompContext *comp_ctx, AOTFuncType *func_type)
 {
-    /* param count + result count + types */
-    return (uint32)sizeof(uint32) * 2 + func_type->param_count
+    /* type flags + parent type idx + is_sub_final + param count + result count
+     * + types */
+    return (uint32)sizeof(uint16) * 6 + func_type->param_count
            + func_type->result_count;
 }
+#endif
 
 static uint32
-get_types_size(AOTType **func_types, uint32 func_type_count)
+get_func_type_info_size(AOTCompContext *comp_ctx, AOTCompData *comp_data)
```

Comment:
It may include struct/array type info? Had better rename it, e.g. get_type_info_size or get_defined_type_info_size?

Reply:
Will handle it in another PR.